### PR TITLE
Add CSV export for followers

### LINKED ISSUE #22966: CSV export of followers
### Pitch

Add CSV export of followers to the same settings page that allows export of following, mutes, blocks, bookmarks, etc.

This is not intended for re-import or account migrations, and requires very little additional code, since it's almost symmetric with exporting the accounts the user is following.

### Motivation

One of my users asked for this so she could do some offline analysis of her own followers list, and I was surprised it wasn't already there. Anyone looking to do similar analysis, or to back up the exact state of their account, might want this. It doesn't seem to be in the takeout archive already.

### PR DESCRIPTION
**Pull request by VyrCossont**
Fixes #22966.

### CODE REVIEW

File: app/models/export.rb [hunk 1]
```diff
@@ -37,6 +37,14 @@ def to_following_accounts_csv
     end
   end
 
+  def to_follower_accounts_csv
+    CSV.generate(headers: ['Account address'], write_headers: true) do |csv|
+      account.passive_relationships.includes(:account).reorder(id: :desc).each do |follower|
```
Comment (Gargron) [inline app/models/export.rb:42]: You need to do find_each here, you could be dealing with over 300,000 rows here.

**Review by Gargron — CHANGES_REQUESTED** [review]
My main concern is that the number of rows in such a document is practically unbounded, it could easily reach half a million, which has implications on memory usage of the process as well as potential request timeout

File: app/models/export.rb (hunk 1 above)
Comment (VyrCossont) [reply app/models/export.rb:42]: Thanks for the quick feedback, I'll switch this to the `.find_each` batch interface.

**Comment by VyrCossont** [comment]
As for the CSV itself… let's say we're exporting a million followers, times 30 bytes max per username + 253 bytes for the max length of a domain name + 2 byte for the @ and newline, and assuming we're not precompressing any of this or streaming it using chunked transfer encoding, okay, 280 MB is a bit much to have in RAM, we might want a temp file and a background job for this. I'll look at the archive export code.

**Comment by VyrCossont** [comment]
I think I'm the only admin that needs this, and I've already applied the small-follower-list version to my own instance, so I'm just going to close this out.

**Pull request closed without merging**